src/uu/pidof/src/pidof.rs

```diff
@@ -0,0 +1,161 @@
+
+use std::path::PathBuf;
+
+use clap::{crate_version, Arg, ArgAction, ArgMatches, Command};
+use uu_pgrep::process::{walk_process, ProcessInformation};
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+const ABOUT: &str = help_about!("pidof.md");
+const USAGE: &str = help_usage!("pidof.md");
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let arg_program_name = matches.get_many::<String>("program-name");
+    let arg_separator = matches.get_one::<String>("d").unwrap();
+
+    if arg_program_name.is_none() {
+        uucore::error::set_exit_code(1);
+        return Ok(());
+    };
+
+    let collected = collect_matched_pids(&matches);
+
+    if collected.is_empty() {
+        uucore::error::set_exit_code(1);
+        return Ok(());
+    };
+
+    let output = collected
+        .into_iter()
+        .map(|it| it.pid.to_string())
+        .collect::<Vec<_>>()
+        .join(arg_separator);
+
+    let flag_quiet = matches.get_flag("q");
+    if !flag_quiet {
+        println!("{output}");
+    }
+
+    Ok(())
+}
+
+fn get_executable_name(process: &mut ProcessInformation) -> String {
+    let binding = process.cmdline.split(' ').collect::<Vec<_>>();
+    let mut path = binding.first().unwrap().to_string();
+
+    if path.is_empty() {
+        path.clone_from(&process.status()["Name"]);
+    };
+
+    PathBuf::from(path)
+        .file_name()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_string()
+}
+
+fn collect_matched_pids(matches: &ArgMatches) -> Vec<ProcessInformation> {
+    let program_name: Vec<_> = matches
+        .get_many::<String>("program-name")
+        .unwrap()
+        .cloned()
+        .collect();
+
+    let collected = walk_process().collect::<Vec<_>>();
+    let arg_omit_pid = matches
+        .get_many::<usize>("o")
+        .unwrap_or_default()
+        .copied()
+        .collect::<Vec<_>>();
+
+    let mut processed = Vec::new();
+    for mut process in collected {
+        let contains = program_name.contains(&get_executable_name(&mut process));
+        let should_omit = arg_omit_pid.contains(&process.pid);
+
+        if contains && !should_omit {
+            processed.push(process)
+        }
+    }
+
+    processed.sort_by(|a, b| b.pid.cmp(&a.pid));
+
+    let flag_s = matches.get_flag("s");
+    if flag_s {
+        match processed.first() {
+            Some(first) => vec![first.clone()],
+            None => Vec::new(),
+        }
+    } else {
+        processed
+    }
+}
+
+#[allow(clippy::cognitive_complexity)]
+pub fn uu_app() -> Command {
+    Command::new(uucore::util_name())
+        .version(crate_version!())
+        .about(ABOUT)
+        .override_usage(format_usage(USAGE))
+        .infer_long_args(true)
+        .arg(
+            Arg::new("program-name")
+                .help("Program name.")
+                .required(true)
+                .index(1),
```

Comment:
I think some setting is missing for this arg because currently I get an "unexpected argument" error when using multiple program names like `cargo run pidof bash bash bash`.